File: C# for VB/3-csharp-visual-basic-dotnet-developers-m3-exercise-files/demos/vb/VBObjects/Message.vb
Imports System.Text

Public Class Message

    Private _text As String
    Public Property Text() As String
        Get
            Return _text
        End Get
        Set(ByVal value As String)
            _text = value
        End Set
    End Property

    Public Function Save(ByRef status As String)
        status = "Success"
        Return True
    End Function

    Public Sub Concatenate(delmiter As String, ParamArray items() As String)
        Dim result As New StringBuilder()
        For Each item As String In items
            result.Append(item & delmiter)
        Next

        Me._text = result.ToString().TrimEnd(delmiter)
    End Sub
End Class

File: C# for VB/3-csharp-visual-basic-dotnet-developers-m3-exercise-files/demos/cs/CSObjects/Message.cs
using System;
using System.Text;

namespace CSObjects
{
    public class Message
    {
        public string Text { get; set; }

        public bool Save(out string status)
        {
            status = "Success";
            return true;
        }

        public void Concatenate(string delimiter, params string[] words)
        {
            StringBuilder result = new StringBuilder();

            foreach (string word in words)
            {
                result.Append(word + delimiter);
            }

            this.Text = result.ToString().TrimEnd(delimiter.ToCharArray());
        }
    }
}
